### PR TITLE
💅Resolves warnings from "bundle exec rake test:unit" output

### PR DESCRIPTION
**Pull request by andriantoniades**
Addresses Issue #156, this resolves warnings from output as a result of running `bundle exec rake test:unit.`

### CODE REVIEW

File: lib/fastly/service.rb [hunk 1]
```diff
@@ -2,7 +2,8 @@
 class Fastly
   # Represents something you want to serve - this can be, for example, a whole web site, a Wordpress site, or just your image servers
   class Service < Base
-    attr_accessor :id, :customer_id, :name, :comment, :versions
+    attr_accessor :id, :customer_id, :name, :comment
+    attr_writer :versions
```
Comment (gschorkopf) [inline lib/fastly/service.rb:6]: 👍

File: lib/fastly/belongs_to_service_and_version.rb [hunk 2]
```diff
@@ -16,9 +16,9 @@ def version
     end
 
     # Get the number of the Version this belongs to
-    def version_number # rubocop:disable all
-      @version
-    end # rubocop:enable all
+    def version_number
+      @version ||= nil
```
Comment (gschorkopf) [inline lib/fastly/belongs_to_service_and_version.rb:20]: What's the warning here? An undefined instance variable will already point to `nil`.

```ruby
> @foobar 
=> nil
> @foobar || = nil
> @foobar 
=> nil

File: test/fastly/client_test.rb [hunk 3]
```diff
@@ -115,6 +115,9 @@
         '/service/blah',
         {include_auth: false}
       )
+
+      assert_equal resp.class, Hash
+      assert_includes resp, "i"
```
Comment (gschorkopf) [inline test/fastly/client_test.rb:120]: I see other tests are making this assertion, but what does it mean? I see `JSON.generate(i: "dont care")` above, maybe that's why we're asserting `"i"` is present?

File: lib/fastly/belongs_to_service_and_version.rb (hunk 2 above)
Comment (gschorkopf) [reply lib/fastly/belongs_to_service_and_version.rb:20]: Also, memoizing an instance var in a public method and not in `initialize` is a _little_ surprising.

Comment (annawinkler) [reply lib/fastly/belongs_to_service_and_version.rb:20]: @andriantoniades could you please post the warning here? We talked about an `initialize` method or doing the memoize, and I don't have a strong preference either way. If using `initialize` is more standard, we can switch to that.

Comment (andriantoniades) [reply lib/fastly/belongs_to_service_and_version.rb:20]: Sure! The warning was this:
`warning: instance variable @version not initialized`

File: test/fastly/client_test.rb (hunk 3 above)
Comment (andriantoniades) [reply test/fastly/client_test.rb]: Good catch, this was an unintentional addition, which I removed in my latest commit.

File: lib/fastly/belongs_to_service_and_version.rb (hunk 2 above)
Comment (andriantoniades) [reply lib/fastly/belongs_to_service_and_version.rb:20]: This is the same style as the two methods above it. If we're choosing a different style, then this is going to be a bigger change as we'd want to change it everywhere. So for now, I'm going to use this to clear the warning and we can have a bigger discussion about memoization in a public method with the broader team.

**Pull request merged**